src/download.rs

```diff
@@ -81,13 +85,13 @@ pub fn download_file(url: &str, file_path: &Path, redirect_path: &Path) -> Resul
           measurements.pop_front();
         }
         let avg_speed = measurements.iter().sum::<f64>() / measurements.len() as f64;
-        let eta = if avg_speed > 0.0 {
+        let eta = if avg_speed > 1.0 {
           (total_size as f64 - downloaded as f64) / avg_speed
         } else {
           0.0
         };
 
-        let progress = (downloaded as f64 / total_size as f64 * 100.0).round() as i64;
+        let progress = utils::to_precision(downloaded as f64 / total_size as f64 * 100.0, 2);
         if progress > last_reported_progress {
```

Comment:
I think it would be easier to do something like this instead (put desireable reporting frequency in place of `0.001` (1 == 100%):
```rust
let progress = downloaded as f64 / total_size as f64;
if progress > last_reported_progress  + 0.001 {
```